src/roles/hostname_check/files/foreman-hostname-check

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+RED=$(tput setaf 1 2>/dev/null)
+GREEN=$(tput setaf 2 2>/dev/null)
+RESET=$(tput sgr0 2>/dev/null)
+
+EXIT_CODE=0
+
+function success() {
+    echo -e "\n${GREEN}[OK]${RESET}\n"
+}
+
+function error() {
+    echo -e "\n${RED}[FAIL]${RESET}\n"
+    CURRENT_EXIT_CODE=$1
+    EXIT_CODE=$((EXIT_CODE|CURRENT_EXIT_CODE))
+    echo -e "$2" >&2
+}
+
+function check-not-localhost() {
+    printf "Checking that hostname is not 'localhost' or empty: "
+    HOSTNAME=$(hostname)
+    if [[ "$HOSTNAME" == "localhost" || "$HOSTNAME" == "localhost.localdomain" || -z "$HOSTNAME" ]]; then
+        error 1 "Hostname is invalid: '$HOSTNAME'"
+    else
+        success
+    fi
+}
+
+function check-fqdn-resolves() {
+    printf "Checking that FQDN resolves via DNS or /etc/hosts: "
+    FQDN=$(hostname -f)
```

Comment:
Puppet and Facter used `hostname -f` a lot. In our new installer we don't. Perhaps it's better to pass them as explicit inputs. We use `ansible_fqdn` everywhere so that's all you need to care about.